src/server/pps/server/api_server.go

```diff
@@ -2770,7 +2770,7 @@ func (a *apiServer) deletePipeline(pachClient *client.APIClient, request *pps.De
 	if err := a.jobs.ReadOnly(ctx).GetByIndex(ppsdb.JobsPipelineIndex, request.Pipeline, jobPtr, col.DefaultOptions, func(jobID string) error {
 		eg.Go(func() error {
 			_, err := a.DeleteJob(ctx, &pps.DeleteJobRequest{Job: client.NewJob(jobID)})
-			if isNotFoundErr(err) {
+			if isNotFoundErr(err) || auth.IsErrNoRoleBinding(err) {
```

Comment:
This will never succeed with auth enabled since the repo was already deleted. AFAICT that's not a regression in the new auth system, we've been silently ignoring it forever. It seems like long-term we should move the job stopping logic to pollPipelines as the comment suggests?

Reply:
...whoops :grimacing:

re:moving the logic, TBH I'm not totally sure I understand, but I do remember writing the comment. FWIW, basically, the issue is that this operates on the jobs that exist right now, but if there are any output commits that don't have a job yet (the delay being a product of the fact the it's the worker master, running elsewhere, that propagates output commits -> jobs) then a new job could show up even after all existing jobs are deleted.

The idea was basically that `pollPipelines` could list all jobs and delete any job where the job's pipeline doesn't exist anymore, though I think this logic is compatible with that change—i.e. we could make that change and leave this as well if we wanted.